test/package-managers-tests/basic-pnpm/test/index.test.js

```diff
@@ -0,0 +1,74 @@
+/* eslint-env jest */
+import { execSync } from 'child_process'
+import fs from 'fs-extra'
+import path from 'path'
+
+jest.setTimeout(100 * 1000)
+
+beforeAll(async () => {
+  await clean()
+})
+
+afterAll(async () => {
+  await clean()
+})
+
+test('pnpm installs', async () => {
+  const pnpm = getStdout('yarn bin pnpm')
+  exec(pnpm + ' init -y')
+  exec(pnpm + ' add next react react-dom')
```

Comment:
There is no way to do that other than modifying the CI script, making it even more brittle

Furthermore, the test installs the dependencies before starting, the installed node_modules will always have precedence over the root node_modules

Reply:
That's fair.

I'm more worried about cases where a dependency is used by the Next.js app under test, but is missing from the pnpm `node_modules` tree.

For example, if we forget to install `react` or another peer dependency, then it will instead be imported from the monorepo `node_modules`.

A more likely scenario is that a dependency is mistakenly omitted from the `next` package's dependencies list, e.g. https://github.com/vercel/next.js/pull/8344 (this example is for "next-server", but the same could apply to "next"). We want this to cause the pnpm integration test to fail, but if it's not isolated from the monorepo `node_modules`, then the test will pass.

Reply:
> There is no way to do that other than modifying the CI script

Is it possible to have this test create a temp folder outside of the next.js project tree, copy over package.json and pages and run the pnpm commands on that folder?

Reply:
yes i can do that